service/gcs/core/gcs/networking.go

```diff
@@ -18,7 +19,7 @@ import (
 // namespace and configures it there.
 func (c *gcsCore) configureAdapterInNamespace(container runtime.Container, adapter prot.NetworkAdapter) error {
 	id := adapter.AdapterInstanceID
-	interfaceName, err := c.instanceIDToName(id)
+	interfaceName, err := instanceIDToName(c.OS, id)
```

Comment:
I'm confused - why is the OS needed here?

Reply:
Because `instanceIDToName` issued a `c.OS.ReadDir` but we need to call this from V2 which doesn't have `(c *gcsCore)` but does have `oslayer.OS`. So instead we just pass the part of `*gcsCore` we need and then V2 can do the same.